compiler-rt/lib/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cpp

```diff
@@ -403,7 +403,48 @@ struct ScopedSetTracerPID {
   }
 };
 
+// This detects whether ptrace is blocked (e.g., by seccomp), by forking and
+// then attempting ptrace.
+// This separate check is necessary because StopTheWorld() creates a *thread*,
+// and therefore cannot use waitpid() due to the shared errno.
```

Comment:
`errno` should be thread local on any reasonable platform?

Reply:
Thanks, updated.

Reply:
> maybe the real reason is the clone syscall?

*maybe the real treasure was the clones we made along the way